### ISSUE TITLE
PHP: Inline-Mode doesn't handle closing PHP-Tags properly

### ISSUE THREAD

**Issue by mutec** [ajaxorg/ace#3170]
If you set `inline: true`, PHP highlighting starts right from the beginning, but if you close the PHP-section/tag and try writing plain HTML, you'll still get a PHP-Highlighting (instead of un-highlighted text):
![image](https://cloud.githubusercontent.com/assets/1749344/21295680/c87d1aa6-c55a-11e6-9b3f-69ff37a93e2f.png)
see also: https://jsfiddle.net/b0fqo0tj/

**Comment by nightwing** [comment]
This was the intended behavior since inline mode is just for php, without any closing and opening tags.
Could you tell how/where do you use code like this? Maybe we need another mode for php.

**Comment by mutec** [comment]
I'm developing a CMS which provides the possibility to add custom PHP-contents which are edited via the CMS's control panel. As far as I know are some customers using self-designed forms which are based on a mix of PHP and HTML (or Smarty). For highlighting the HTML-code of those forms typing HTML outside the PHP-Section would be better (and can't cause problems escaping the `'` or `"`).
For example the code from above: https://test.mysterycode.de/vortex/cms/index.php?wsctest/
The user-list is created by php and the blue info-box is displayed outside the PHP-tag.

In a nutshell: I'd need ACE's PHP-highlighting like it is (parsing closing and opening tags) but starting the highlighting from the first character to the closing PHP-Tag and then from another opening to closing tag (and so on).
I hope you got what I wanted to describe. :)

**Comment by mutec** [comment]
@nightwing Did you think about that yet or should I try to find a custom solution?

**Comment by nightwing** [comment]
one possible solution is to find a way of passing  `"php-start"` to  https://github.com/ajaxorg/ace/blob/v1.2.6/lib/ace/tokenizer.js#L239 instead of using hardcoded `"start"` but i am not sure how to do it properly.
